src/NHibernate.Test/Async/SystemTransactions/ResourceManagerFixture.cs

```diff
@@ -0,0 +1,186 @@
+				}
+				_log.Info("Scope disposed");
+				while (eventTransaction == null)
+					await (Task.Delay(10));
+				_log.Info("Event transaction received");
+				Assert.That(eventTransaction, Is.SameAs(clone));
+			}
+			finally
+			{
+				clone?.Dispose();
+			}
+
+			void Clone_TransactionCompleted(object sender, TransactionEventArgs e)
+			{
+				eventTransaction = e.Transaction;
+			}
+		}
+
+		#endregion
+		#region Rollback
+
+		[Test]
+		public async Task DistributedTransactionStatusFromCompletionEventShouldNotBeActiveOnRollbackAsync()
+		{
+			SysTran clone = null;
+			SysTran eventTransaction = null;
+			TransactionStatus? cloneStatusAtCompletion = null;
+			try
+			{
+				using (CreateDistributedTransactionScope())
+				{
+					_log.InfoFormat(
+						"Scope opened, id {0}, distributed id {1}",
+						SysTran.Current.TransactionInformation.LocalIdentifier,
+						SysTran.Current.TransactionInformation.DistributedIdentifier);
+					clone = SysTran.Current.Clone();
+					clone.TransactionCompleted += Clone_TransactionCompleted;
+					_log.Info("Scope not completed");
+				}
+				_log.Info("Scope disposed");
+				while (eventTransaction == null)
+					await (Task.Delay(10));
+				_log.Info("Event transaction received");
+				Assert.That(cloneStatusAtCompletion, Is.Not.EqualTo(TransactionStatus.Active));
+			}
+			finally
+			{
+				clone?.Dispose();
+			}
+
+			void Clone_TransactionCompleted(object sender, TransactionEventArgs e)
+			{
+				cloneStatusAtCompletion = FailsafeGetTransactionStatus(clone);
+				eventTransaction = e.Transaction;
+			}
+		}
+
+		#endregion
+		#endregion
+		#region Non distributed
+		#region Commit
+
+		#endregion
+		#region In-doubt
+
+		#endregion
+		#region Rollback
+
+		#endregion
+
+		#region Failure
+
+		[Test]
+		[Explicit("Failing")]
+		public async Task TransactionFromCompletionEventShouldBeTheOneToWhichTheEventIsAttachedAsync()
+		{
+			SysTran clone = null;
+			SysTran eventTransaction = null;
+			try
+			{
+				using (new TransactionScope(TransactionScopeAsyncFlowOption.Enabled))
+				{
+					_log.InfoFormat(
+						"Scope opened, id {0}, distributed id {1}",
+						SysTran.Current.TransactionInformation.LocalIdentifier,
+						SysTran.Current.TransactionInformation.DistributedIdentifier);
+					clone = SysTran.Current.Clone();
+					clone.TransactionCompleted += Clone_TransactionCompleted;
+					_log.Info("Scope not completed");
+				}
+				_log.Info("Scope disposed");
+				while (eventTransaction == null)
+					await (Task.Delay(10));
+				_log.Info("Event transaction received");
+				Assert.That(eventTransaction, Is.SameAs(clone));
+			}
+			finally
+			{
+				clone?.Dispose();
+			}
+
+			void Clone_TransactionCompleted(object sender, TransactionEventArgs e)
+			{
+				eventTransaction = e.Transaction;
+			}
+		}
+
+		#endregion
+		#endregion
+		#region Tests setup/teardown/utils
+
+		[TearDown]
+		public async Task TearDownAsync(CancellationToken cancellationToken = default(CancellationToken))
```

Comment:
@maca88 methods wit `TearDownAttribute` are specified as being copied. Why did it generate the Async?

Reply:
This is related to [this issue](https://github.com/maca88/AsyncGenerator/issues/68). Copying a method for a partial class would cause a compilation error as the method would be duplicated. Probably, what you want here is to mark all classes with `ExplicitAttribute` to be generated as a new type.

Reply:
Ok, the solution was to apply `[TestFixture]` attribute. Thanks for the tip.